blog/2015/09/thinking-about-teaching.html

```diff
@@ -0,0 +1,360 @@
+      than someone who is new to ideas like "variable", "list", and "function".
+    </p>
+  </dd>
+  <dt>
+    Roughly half of the respondents don't like the way we jump right
+    into an example.
+  </dt>
+  <dd>
+    <p>
+      The intention was to get to an authentic task as quickly as possible,
+      and "load CSV, create plot" is something every scientist does.
+      I still think it works for people who are learning Python as a second language,
+      but if that's only 1/3 of our workshops,
+      it's not surprising that it often doesn't work.
+    </p>
+  </dd>
+  <dt>
+    Only 1/3 of respondents got through the whole Python lesson.
+  </dt>
+  <dd>
+    <p>
+      Which means we need to cut material&mdash;even if we say,
+      "It's OK to go as far as you can,"
+      instructors will often feel that they've let their audience down
+      if they don't get through it all.
+    </p>
+  </dd>
+  <dt>
+    Over 80% of the respondents are teaching our materials with little
+    or no modification, but 22% are using an older version.
+  </dt>
+  <dd>
+    <p>
+      Based on a handful of conversations,
+      I think people use older material because
+      (a) they stick to whatever lesson they first taught,
+      and/or (b) they don't like the NumPy-early approach of the current lesson.
+      I don't know what we can do about (a),
+      but our "Python as a first language" lesson
+      should start with simpler tasks.
+    </p>
+  </dd>
+  <dt>
+    The defensive programming section which is skipped by over 85% of
+    the people who answered the survey, so we should move this
+    material elsewhere.
+  </dt>
+  <dd>
+    <p>
+      I (reluctantly) agree,
+      at least for people who are learning programming for the first time.
+      As with <a href="{{page.root}}/blog/2014/10/why-we-dont-teach-testing.html">testing</a>,
+      there's no point putting scarce classroom or maintenance time into something that doesn't stick.
+    </p>
+  </dd>
+  <dt>
+    82% of the people answering the survey are teaching out of the
+    Jupyter Notebook.
+  </dt>
+  <dd>
+    <p>
+      I was surprised by that, but I now wonder if it explains why it
+      takes so long to get through material with novices, who are
+      facing the double cognitive load of a new programming language
+      <em>and</em> a new kind of editing interface.  I teach the
+      old-fashioned way (scripts run from the command line) to avoid
+      this extra load, but clearly it's time for me to change.
+    </p>
+  </dd>
+</dl>
+<p>
+  Stepping back,
+  the second-greatest benefit of teaching in the large is that
+  we can pool our experiences.
+  (The greatest is having someone who can genuinely sympathize when something has gone wrong...)
+  It would take even our most active instructors several years to learn this much about our lessons;
+  comparing notes helps us all improve faster.
```

Comment:
Can you add a link to the survey and tell people it not to late to add their own data?